src/Cardano/Wallet/Network.hs

```diff
@@ -18,29 +27,51 @@ import Cardano.Wallet.Primitive.Types
 import Control.Concurrent
     ( threadDelay )
 import Control.Exception
-    ( Exception, throwIO )
+    ( Exception )
 import Control.Monad.IO.Class
     ( MonadIO, liftIO )
 import Control.Monad.Trans.Except
     ( ExceptT, runExceptT )
+import Data.Text
+    ( Text )
 import Data.Time.Units
     ( Millisecond, toMicroseconds )
+import GHC.Generics
+    ( Generic )
 
-data NetworkLayer m e0 e1 = NetworkLayer
-    { nextBlocks :: SlotId -> ExceptT e0 m [Block]
+data NetworkLayer m = NetworkLayer
+    { nextBlocks :: SlotId -> ExceptT ErrNetworkUnreachable m [Block]
         -- ^ Gets some blocks from the node. It will not necessarily return all
         -- the blocks that the node has, but will receive a reasonable-sized
         -- chunk. It will never return blocks from before the given slot. It
         -- may return an empty list if the node does not have any blocks from
         -- after the starting slot.
 
     , networkTip
-        :: ExceptT e1 m (Hash "BlockHeader", BlockHeader)
+        :: ExceptT ErrNetworkUnreachable m (Hash "BlockHeader", BlockHeader)
+        -- ^ Get the current network tip from the chain producer
 
     , postTx
-        :: SignedTx -> ExceptT e1 m ()
+        :: SignedTx -> ExceptT ErrPostTx m ()
+        -- ^ Broadcast a transaction to the chain producer
     }
 
+-- | Network is not reachable
+newtype ErrNetworkUnreachable
+    = ErrNetworkUnreachable Text
+    deriving (Generic, Show, Eq)
+
+instance Exception ErrNetworkUnreachable
+
+-- | Error while trying to send a transaction
+data ErrPostTx
+    = ErrPostTxNetworkUnreachable ErrNetworkUnreachable
```

Comment:
This still kind of hurts me a bit :/ (even though we don't have any better alternative)

Reply:
We do have an alternative, but it's a trade-off between boilerplate and complexity we have here :/ ... 

Have a look at: https://www.parsonsmatt.org/2018/11/03/trouble_with_typed_errors.html